### PR TITLE
[Custom RPCs] Allow displaying and removing custom networks

### LINKED ISSUE #2758: Show previously added networks on the `Custom Networks` page.
Depends on https://github.com/tallycash/extension/issues/2757

The `delete` icon should remove the chain from the wallet (This can be tackled in another issue if desired).

image

### PR DESCRIPTION
**Pull request by hyphenized**
Closes #2758

Allows managing custom networks from within the settings page

thismightjustbethenextbigthing

## Testing Env

```
SUPPORT_CUSTOM_NETWORKS=true
SUPPORT_CUSTOM_RPCS=true
```

## To Test
- [ ] Add a few custom networks
- [ ] Head to settings, try removing some of the networks just added

Latest build: [extension-builds-3032](https://github.com/tahowallet/extension/suites/11529113568/artifacts/596372116) (as of Mon, 13 Mar 2023 16:40:02 GMT).

### CODE REVIEW

File: background/redux-slices/networks.ts [hunk 1]
```diff
@@ -50,14 +55,41 @@ const networksSlice = createSlice({
           block?.baseFeePerGas ?? null
       }
     },
+    /**
+     * Receives all supported networks as the payload
+     */
     setEVMNetworks: (immerState, { payload }: { payload: EVMNetwork[] }) => {
```
Comment (kkosiorowska) [inline background/redux-slices/networks.ts:61]: Maybe we could add it here. 🤔

**Review by kkosiorowska — COMMENTED** [review]
Looks good! I think it would be nice to add some tests.

**Comment by kkosiorowska** [comment]
We should also improve the display of network names.  

Screenshot 2023-02-27 at 10 54 33

File: background/redux-slices/networks.ts [hunk 2]
```diff
@@ -50,14 +55,41 @@ const networksSlice = createSlice({
...
+      const chainIds = payload.map((network) => network.chainID)
+
       payload.forEach((network) => {
         immerState.evmNetworks[network.chainID] = network
       })
+
+      // Remove payload missing networks from state
```
Comment (0xDaedalus) [inline background/redux-slices/networks.ts:68]: 📈

**Review by 0xDaedalus — DISMISSED** [review]

**Review by 0xDaedalus — APPROVED** [review]

**Pull request merged**